spec/features/competitions_list_spec.rb

```diff
@@ -15,16 +15,18 @@
       before do
         visit "/competitions?show_admin_details=yes"
 
-        # The delegate dropdown should have three items: The generic 'None'
-        #   as well as two Delegates created by the `let(:competition)` above.
-        expect(page).to have_css("#delegate div.item", visible: :hidden, count: 3)
```

Comment:
I'm inclined to allow an exception here, because this is our undercover way of assuring _before each individual test_ that the async "load available Delegates" `useQuery` finished loading...

Reply:
Notably, it's *not* an individual test that is supposed to check "can it load". It is an actual guarantee for all tests that it did indeed load -- without this guarantee, all further assumptions about "selecting a Delegate from the dropdown" don't make much sense

Reply:
ah okay. I couldn't infer that from the comment

Reply:
the tests still pass interestingly

Reply:
That's just a happy coincidence. It means that during _this one particular test run_, the loading happened to be fast enough for the dropdown to be populated already. So it's literally luck. With the assumption, you can be _sure_ that the dropdown loaded.